src/jdk.hotspot.agent/macosx/native/libsaproc/MacosxDebuggerLocal.m

```diff
@@ -277,8 +290,10 @@ jlong lookupByNameIncore(
 
   jlong address = 0;
 
-JNF_COCOA_ENTER(env);
-  NSString *symbolNameString = JNFJavaToNSString(env, symbolName);
+  NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
+  @try {
```

Comment:
Although there are only 3 places where the `JNF_COCOA_ENTER/EXIT` macros are used, it seems it would still be worth taking the same approach you did for `java.desktop` and add the replacement macros instead of inlining them. So just copy what you added to `src/java.desktop/macosx/native/libosxapp/JNIUtilities.h`, and replace `JNF_COCOA_ENTER` with `JNI_COCOA_ENTER/EXIT`. Otherwise the `JNF_COCOA_ENTER/EXIT` changes look fine to me, but I'm just basing this on a comparison with what you've done with `java.desktop`. I'm no expert in this area.

Reply:
OK .. I don't really mind either way and if this helps gets it pushed .. so I've updated.